airflow/migrations/versions/0140_2_9_0_update_trigger_kwargs_type.py

```diff
@@ -38,13 +41,43 @@
 airflow_version = "2.9.0"
 
 
+def get_session() -> sa.orm.Session:
+    conn = op.get_bind()
+    sessionmaker = sa.orm.sessionmaker()
+    return sessionmaker(bind=conn)
+
 def upgrade():
-    """Update trigger kwargs type to string"""
+    """Update trigger kwargs type to string and encrypt"""
     with op.batch_alter_table("trigger") as batch_op:
         batch_op.alter_column("kwargs", type_=sa.Text(), )
 
+    if not context.is_offline_mode():
+        session = get_session()
+        try:
+            for trigger in session.query(Trigger):
+                trigger.kwargs = trigger.kwargs
+            session.commit()
+        finally:
+            session.close()
+
 
 def downgrade():
-    """Unapply update trigger kwargs type to string"""
+    """Unapply update trigger kwargs type to string and encrypt"""
+    if context.is_offline_mode():
+        print(dedent("""
+        ------------
+        --  WARNING: Unable to decrypt trigger kwargs automatically in offline mode!
```

Comment:
I don't think we can do any better than spitting out a warning :(.

But if folks have ideas here, happy to hear them.

Reply:
It's a moving target. There may be none now but when they go run the generated sql it blows up.

Reply:
This is what it looks like in the generated sql:

```
BEGIN;

-- Running downgrade 1949afb29106 -> ee1467d4aa35


------------
--  WARNING: Unable to decrypt trigger kwargs automatically in offline mode!
--  If any trigger rows exist when you do an offline downgrade, the migration will fail.
------------

ALTER TABLE trigger ALTER COLUMN kwargs TYPE JSON USING kwargs::json;

UPDATE alembic_version SET version_num='ee1467d4aa35' WHERE alembic_version.version_num = '1949afb29106';


```